resources/2024-06-support-secretstores-env.md

```diff
@@ -0,0 +1,417 @@
+that are not part of this feature (dependencies). This diagram generally
+treats the components as black boxes. Provide a pointer to a more detailed
+design document, if one exists. 
+-->
+The design of this new feature will require updates to the versioned datamodel, the conversion functions, the containers typespec and the common typespec.
+These will be breaking changes to the schema. Users will need to update the environment variables in their bicep files to use the new secret reference type. 
+
+### Architecture Diagram
+![Architecture Diagram](./2024-06-support-secretstores-env/secretstores-env.png)
+
+### Detailed Design
+
+<!--
+This section should be detailed and thorough enough that another developer
+could implement your design and provide enough detail to get a high confidence
+estimate of the cost to implement the feature but isn’t as detailed as the 
+code. Be sure to also consider testability in your design.
+
+For each change, give each "change" in the proposal its own section and
+describe it in enough detail that someone else could implement it. Cover
+ALL of the important decisions like names. Your goal is to get an agreement
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+Discuss the rationale behind architectural choices and alternative options 
+considered during the design process.
+-->
+The design of this feature will require updates to the versioned datamodel, the conversion functions, the containers typespec and the common typespec to leverage the new secret reference type and provide support for secret stores in environment variables beyond the current support for environment variables with a string value.
+
+#### Advantages (of each option considered)
+<!--
+Describe what's good about this plan relative to other options. 
+Provides better user experience? Does it feel easy to implement? 
+Provides flexibility for future work?
+-->
+Advantages of this approach are that it allows users to provide secrets to a container through environment variables. This is a common use case and will make it easier for users to provide secrets to their containers. In using much of the existing functionality of Radius, this approach is also relatively simple to implement.
+
+#### Disadvantages (of each option considered)
+<!--
+Describe what's not ideal about this plan. Does it lock us into a 
+particular design for future changes or is it flexible if we were to 
+pivot in the future. This is a good place to cover risks.
+-->
+Disadvantages are that it will break existing bicep files that use environment variables. Users will need to update their bicep files to use the new secret reference type.
```

Comment:
I want to mention that I'm personally on board with the breaking change. The current design is too naive, and I've expected a breaking change here for a long time.